overlord/snapstate/snapstate.go

```diff
@@ -1273,6 +1276,16 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []*s
 		updated = append(updated, name)
 	}
 
+	// Report a useful error if nothing can be updated and we have at
+	// least one auto-refresh change in progress.
+	if refreshAll && len(tasksets) == 0 {
+		for _, err := range updateErrs {
+			if _, ok := err.(*ChangeConflictError); ok {
+				return nil, nil, fmt.Errorf("auto-refresh has conflicting changes in progress")
```

Comment:
should this be some kind of ChangeConflictError again? or you think that's confusing?

Reply:
I think that's an excellent suggestion! I will add this